FuelLabs/fuel-indexer
# enhancement: add resume index functionality

Closes #568 

## Changelog
• adds `resumable` to manifest
• creates a new queries:: method `last_block_height_for_indexer`

## Testing 
- Run 
`cargo run --bin fuel-node`
`cargo run --bin web-api`
Update `fuel_indexer_test.yaml` to be resumable by adding the field:
```
resumable: true
```
- Run the test indexer
`cargo run --bin fuel-indexer -- run --manifest packages/fuel-indexer-tests/components/indices/fuel-indexer-test/fuel_indexer_test.yaml`

-Fire off some events (more than 2)
`curl -v -X POST http://127.0.0.1:8000/ping`

- Quit the indexer process. 
- Start it again with the same command above. Search the logs for `Last block`. It should be more than 1. 

I opted not to write an e2e test here as I wasn't sure yet how to actually quit the indexer and then start it again whilst the api and fuel node were still up. I could figure it out if we still want this.
